pkg/controller/server/device/managed.go

```diff
@@ -118,22 +124,22 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (resource.E
 	}
 
 	// Update device status
-	d.Status.ID = device.ID
-	d.Status.Hostname = device.Hostname
-	d.Status.Href = device.Href
-	d.Status.State = device.State
+	d.Status.AtProvider.ID = device.ID
```

Comment:
Continuing to set the `ID` in status in addition to external name annotation as it is part of the device observation struct.